File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/SetmealService.java
package common.niuniu.service;

import common.niuniu.dto.SetmealDTO;
import common.niuniu.dto.SetmealPageDTO;
import common.niuniu.entity.Setmeal;
import common.niuniu.result.PageResult;
import common.niuniu.vo.DishItemVO;
import common.niuniu.vo.SetmealVO;
import common.niuniu.vo.SetmealWithPicVO;

import java.util.List;

public interface SetmealService {
    void addSetmeal(SetmealDTO setmealDTO);

    PageResult getPageList(SetmealPageDTO setmealPageDTO);

    SetmealVO getSetmealById(Integer id);

    void onOff(Integer id);

    void update(SetmealDTO setmealDTO);

    void deleteBatch(List<Integer> ids);

    List<Setmeal> getList(Integer categoryId);

    List<DishItemVO> getSetmealDishesById(Integer id);

    SetmealWithPicVO getSetmealWithPic(Integer id);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/SetmealService.kt
package common.niuniu.service

import common.niuniu.dto.SetmealDTO
import common.niuniu.dto.SetmealPageDTO
import common.niuniu.po.Setmeal
import common.niuniu.result.PageResult
import common.niuniu.vo.DishItemVO
import common.niuniu.vo.SetmealVO
import common.niuniu.vo.SetmealWithPicVO

interface SetmealService {
    fun addSetmeal(setmealDTO: SetmealDTO?)

    fun getPageList(setmealPageDTO: SetmealPageDTO?): PageResult?

    fun getSetmealById(id: Int?): SetmealVO?

    fun onOff(id: Int?)

    fun update(setmealDTO: SetmealDTO?)

    fun deleteBatch(ids: List<Int?>?)

    fun getList(categoryId: Int?): List<Setmeal>?

    fun getSetmealDishesById(id: Int?): List<DishItemVO>?

    fun getSetmealWithPic(id: Int?): SetmealWithPicVO?
}
